xdsl/rewriter.py

```diff
@@ -119,6 +119,52 @@ def replace_op(
 
         block.erase_op(op, safe_erase=safe_erase)
 
+    @staticmethod
+    def update_op(
+        op: Operation,
+        new_results: Sequence[SSAValue | None] | None = None,
+        safe_erase: bool = True,
+        *,
+        operands: Sequence[SSAValue] | None = None,
+        result_types: Sequence[Attribute] | None = None,
+        properties: Mapping[str, Attribute] | None = None,
+        attributes: Mapping[str, Attribute] | None = None,
+        successors: Sequence[Block] | None = None,
+        regions: Sequence[Region] | None = None,
+    ):
```

Comment:
```suggestion
    @staticmethod
    def rebuild_op_in_place(
        op: Operation,
        operands: Sequence[SSAValue] | None = None,
        result_types: Sequence[Attribute] | None = None,
        properties: Mapping[str, Attribute] | None = None,
        attributes: Mapping[str, Attribute] | None = None,
        successors: Sequence[Block] | None = None,
        regions: Sequence[Region] | None = None,
    ):
```

suggested method name and signature